src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,236 @@
+                                           tr("Other sequences in PCR reaction"),
+                                           wgtOtherSequencesInPcr,
+                                           true);
+
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto seqLength = activeSequenceContext->getSequenceLength();
+
+    sbLeftAreaStart->setValue(1);
+    sbLeftAreaEnd->setValue(seqLength / 10);
+    sbRightAreaStart->setValue(seqLength / 10 * 5);
+    sbRightAreaEnd->setValue(seqLength / 10 * 6);
+
+    connect(pbStart, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_start);
+    connect(tbLeftAreaSelectManually, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually);
+    connect(tbRightAreaSelectManually, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually);
+
+    U2WidgetStateStorage::restoreWidgetState(savableWidget);
+
+    sl_updateMaxValues();
+    for (auto minSb : minMaxValuesSb.keys()) {
+        connect(minSb, &QSpinBox::editingFinished, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_minValueChanged);
+        connect(minMaxValuesSb[minSb], &QSpinBox::editingFinished, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_maxValueChanged);
+    }
+    connect(tbLoadBackbone, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadBackbone);
+    connect(tbsaveRandomSequences, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_saveRandomSequences);
+    connect(tbLoadOtherSequencesInPcr, &QAbstractButton::clicked, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_loadOtherSequenceInPcr);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_start() {
+    //TODO
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_selectManually() {
+    auto currentSMButton = qobject_cast<QToolButton*>(sender());
+    SAFE_POINT(currentSMButton != nullptr, L10N::nullPointerError("QAbstractButton"), );
+
+    if (smButton != nullptr) {
+        bool disableState = true;
+        if (smButton->isChecked()) {
+            smButton->setChecked(false);
+            disableState = false;
+        }
+        smButton = nullptr;
+        start = nullptr;
+        end = nullptr;
+        disconnect(updateRegionConnection);
+        updateRegionConnection = QMetaObject::Connection();
+
+        if (disableState) {
+            return;
+        }
+    }
+
+    smButton = currentSMButton;
+    if (smButton == tbLeftAreaSelectManually) {
+        start = sbLeftAreaStart;
+        end = sbLeftAreaEnd;
+    } else {
+        start = sbRightAreaStart;
+        end = sbRightAreaEnd;
+    }
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
```

Comment:
nit: unneeded null check

Reply:
I prefer to check any pointers, just in case

Reply:
To be consistent: does 'any' means 'all'? 
In this case for any pointer with a life range bigger than a function scope you should have a check before a use. 
Do you want me to review your code this way: that no checks are missed for any pointers?

The standard way of dealing with nulls in different programming languages is to separate nullable and non-nullable values and never check non-nullable for nulls. The owner of the non-nullable value (here AnnotatedDNAView) must guarantee that the pointer can never be set to null.

In our code non-nullable values are documented on the class level and there is no need to check them outside of the class.

Reply:
OK, removed